### PR TITLE
Lesson 2: Jest, cypress

### PR DESCRIPTION
**Pull request by castortroy06**

### CODE REVIEW

File: cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const { defineConfig } = require("cypress");
```
Comment (NikitaUstsiushenka) [inline cypress.config.js:1]: Config looks like empty, I think you can delete it

File: cypress/e2e/counter.cy.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+describe('Counter element check', () => {
+    it('Ensure Counter element has increase and descrease buttons', () => {
+        cy.visit('http://localhost:3000')
```
Comment (NikitaUstsiushenka) [inline cypress/e2e/counter.cy.js]: You can set up base url In cypress config

File: cypress/e2e/genre-select.cy.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+describe('Genre select element check', () => {
+    it('Ensure clicking element makes it active', () => {
+        cy.visit('http://localhost:3000')
```
Comment (NikitaUstsiushenka) [inline cypress/e2e/genre-select.cy.js]: The same here

File: cypress/support/commands.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// ***********************************************
```
Comment (NikitaUstsiushenka) [inline cypress/support/commands.js]: Empty files can be deleted

File: cypress/support/e2e.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// ***********************************************************
```
Comment (NikitaUstsiushenka) [inline cypress/support/e2e.js]: You delete it

File: src/App.js [hunk 6]
```diff
@@ -1,18 +1,24 @@
...
 const genres = ['All', 'Documentary', 'Comedy', 'Horror', 'Crime'];
 
 function App() {
   const [active, setActive] = useState(genres[0]);
-  const callback = (e) => {
-    e.preventDefault();
-    setActive(e.target.textContent);
+  
+  const callback = (value) => {
+    //console.log(value);
```
Comment (NikitaUstsiushenka) [inline src/App.js]: Please remove comments for console.log, thanks!

File: src/App.js [hunk 7]
```diff
@@ -1,18 +1,24 @@
...
+    setActive(value);
+  };
+  
+  const handleSubmit = (value) => {
+    //console.log(value);
```
Comment (NikitaUstsiushenka) [inline src/App.js]: The same here

File: src/components/Counter/Counter.test.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
+// Counter.test.js
```
Comment (NikitaUstsiushenka) [inline src/components/Counter/Counter.test.js]: Please remove this comment

File: src/components/GenreSelect/GenreSelect.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        e.preventDefault();
+        onChange(e.target.textContent);
+    };
+
+    return (
+        <div className="genres">
+            <h2 className='text-center'>Ticket 3: Genres select</h2>
+            {genres.length &&
+                <ul className="genres--list">
+                    {genres.map((genre, key) => {
```
Comment (NikitaUstsiushenka) [inline src/components/GenreSelect/GenreSelect.js]: You can simplify return, using () => ()

File: src/components/SearchForm/SearchForm.js [hunk 10]
```diff
@@ -22,16 +23,17 @@ const SearchForm = () => {
       <form className="text-start" onSubmit={handleSubmit}>
         <label className="form-label" htmlFor="search">Search</label>
         <input name="search" className="form-control" id="search"
+          data-testid="search-input"
           type="text"
           placeholder="Search..."
           value={searchTerm}
-          onChange={handleInputChange}
+          onChange={(e) => { setSearchTerm(e.target.value) }}
```
Comment (NikitaUstsiushenka) [inline src/components/SearchForm/SearchForm.js:29]: Why do we need onChange and onKeyDown actions?

Comment (castortroy06) [reply src/components/SearchForm/SearchForm.js:29]: @NikitaUstsiushenka  I've added onKeyDown action to test form submit on "Enter" key press

**Pull request closed without merging**